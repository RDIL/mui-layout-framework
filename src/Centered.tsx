import * as React from "react"
import Grid from "@material-ui/core/Grid"

export interface CenteredProps {
    children: React.ReactElement
}

const Centered = (props: CenteredProps) => {
    return (
        <Grid container justify="center">
            {props.children}
        </Grid>
    )
}

export default Centered
