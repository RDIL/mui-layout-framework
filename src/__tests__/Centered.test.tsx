import React from "react"
import Centered from "../Centered"
import renderer from "react-test-renderer"
import { Button, Divider, Typography } from "@material-ui/core"

describe("Centered component", () => {
    it("can render the component with empty children", () => {
        const tree = renderer
            .create(
                <Centered>
                    <div />
                </Centered>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("can render the component with advanced children", () => {
        const tree = renderer
            .create(
                <Centered>
                    <div>
                        <Typography variant="h1">Hello!</Typography>
                        <Divider />
                        <Button variant="contained">Press Me</Button>
                    </div>
                </Centered>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
