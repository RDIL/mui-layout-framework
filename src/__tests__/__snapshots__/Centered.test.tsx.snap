// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered component can render the component with advanced children 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div>
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      Hello!
    </h1>
    <hr
      className="MuiDivider-root"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Press Me
      </span>
    </button>
  </div>
</div>
`;

exports[`Centered component can render the component with empty children 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <div />
</div>
`;
